File: src/Client.UWP/Domain/GeoPoint.cs
namespace Client.Domain
{
    /// <summary>
    /// Represents a square on 2D map.
    /// </summary>
    public struct GeoPoint
    {
        /// <summary>
        /// X location of the middle of the square.
        /// </summary>
        public int X;

        /// <summary>
        /// Y location of the middle of the square.
        /// </summary>
        public int Y;

        /// <summary>
        /// Creates a new instance of <see cref="GeoPoint"/>
        /// </summary>
        /// <param name="x">X coordinate.</param>
        /// <param name="y">Y coordinate.</param>
        public GeoPoint(int x, int y)
        {
            X = x;
            Y = y;
        }
    }
}

File: core/src/com/mygdx/game/domain/GeoPoint.java
package com.mygdx.game.domain;

import lombok.Value;

/**
 * Represents a square on 2D map.
 */
@Value
public class GeoPoint
{
    /** X location of the middle of the square. */
    public int x;

    /** Y location of the middle of the square. */
    public int y;

    public GeoPoint top() { return new GeoPoint(x, y + 1);}
    public GeoPoint down() { return new GeoPoint(x, y - 1);}
    public GeoPoint left() { return new GeoPoint(x - 1, y);}
    public GeoPoint right() { return new GeoPoint(x + 1, y);}
    public GeoPoint topLeft() { return new GeoPoint(x - 1, y + 1);}
    public GeoPoint topRight() { return new GeoPoint(x + 1, y + 1);}
    public GeoPoint downLeft() { return new GeoPoint(x - 1, y - 1);}
    public GeoPoint downRight() { return new GeoPoint(x + 1, y - 1);}
}

